book/src/03_3_http_client.md

```diff
@@ -1,35 +1,49 @@
 # A simple HTTP client
 
-In this exercise, we'll retrieve some data over a HTTP connection to the internet.
+In this exercise, we'll write a small http client that retrieves data over a HTTP connection to the internet.
 
 ## Setup
-You can find a prepared project skeleton in `intro/http-client/exercise`. It includes establishing a WiFi connection, but you must configure it to use your network's credentials in `cfg.toml`.
+
+✅ Go to `intro/http-client/exercise` directory.
+
+✅ Open the docs for this project with the following command:
+
+```
+$ cargo doc --open
+```
+
+✅ Add your network credentials to the `cfg.toml`.
+
+✅ Open the prepared project skeleton in `intro/http-client/exercise`. We'll read through the code together.
 
 ## Making a connection
 
 By default only unencrypted HTTP is available, which rather limits our options of hosts to connect to. We're going to use `http://neverssl.com/`.
 
 In `esp-idf`, HTTP client connections are managed by `http::client::EspHttpClient` in the `esp-idf-svc` crate. It implements the `http::client::Client` trait from `embedded-svc`, which defines functions for [HTTP request methods](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods) like `GET` or `POST`.
 
+✅ Create a new `EspHttpClient` with default values.
+
 TODO currently no `docs.rs` documentation for the svc crates (ping Espressif about this, it's a known issue though)
 
-calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. This makes more sense with `POST` and `PUT` but must still be performed with `GET`.
+Calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. This makes more sense with `POST` and `PUT` but must still be performed with `GET`. 
 
 After this optional send step the `Writer` can be turned into a `Response` from which the received server output can be read:
 
 ```Rust
-let request = client.get(...)
+let request = client.get(url.as_ref())?;
```

Comment:
> Or (maybe additionally): explain in English instead of "by a suggestive variable name" ... something like "the `get` method expects a (shared) reference to a URL"

This is better. I'd say there is too little time to figure this out. I did not find it without peeking at the solution and I think introducing it if it is important makes more sense.